### PR TITLE
Fix incorrect guard and other errors

### PR DESCRIPTION
**Pull request by dan-kez**
Some small additions to the script to fix an incorrect guard and how `chpwd_functions` is extended

### CODE REVIEW

File: zsh-pyenv-lazy-load.plugin.zsh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-if type pyenv &> /dev/null; then
+if ! type pyenv &> /dev/null; then
```
Comment (erikced) [inline zsh-pyenv-lazy-load.plugin.zsh:1]: I believe this is correct as is, the script is supposed to exit if `pyenv` is already found?

File: zsh-pyenv-lazy-load.plugin.zsh [hunk 2]
```diff
@@ -9,8 +9,8 @@ _init_pyenv() {
   if [[ ! "$PATH" == */.pyenv/bin* ]]; then
     export PATH="$HOME/.pyenv/bin:$PATH"
   fi
-  eval "$(pyenv init -)"
-  eval "$(pyenv virtualenv-init -)"
+  eval "$(pyenv init - 2>/dev/null)"
+  eval "$(pyenv virtualenv-init - 2>/dev/null)"
```
Comment (erikced) [inline zsh-pyenv-lazy-load.plugin.zsh:13]: Is there a particular reason why you would need to silence `stderr` output from `pyenv`?

**Review by erikced — COMMENTED** [review]
Thank you for your contribution. I am not sure what you are trying to achieve here. The plugin currently assumes that `pyenv` is not in path and is supposed to add and initialize it whenever `pyenv` is called or `.python-version` is found. Looking at the changes it appears that you already have `pyenv` in `$PATH` when the plugin is loaded? Could you please describe your use case?

File: zsh-pyenv-lazy-load.plugin.zsh (hunk 1 above)
Comment (dan-kez) [reply zsh-pyenv-lazy-load.plugin.zsh:1]: Hey @erikced super sorry on this. I was setting up a new computer late at night and for whatever reason misread this. 🤦‍♂️ You're completely correct

**Comment by dan-kez** [comment]
Hey apologies for the poor PR here.

Was setting up a new laptop late at night and saw problems where there were none. This script does work correctly as is. 

Sorry for the trouble!

**Comment by erikced** [comment]
> Hey apologies for the poor PR here.
> 
> Was setting up a new laptop late at night and saw problems where there were none. This script does work correctly as is.
> 
> Sorry for the trouble!

No problems, please do contribute again if you find an issue or need changes to support your use case.

**Pull request closed without merging**